### PR TITLE
Add unit test coverage for nvidiaGPUManager initialization

### PR DESCRIPTION
**Pull request by jiayingz**
Part of #47750 

```release-note
NONE
```

### CODE REVIEW

**Comment by jiayingz** [comment]
CLA should be signed now.

**Review by vishh — APPROVED** [review]
/lgtm

**Comment by cmluciano** [comment]
@k8s-bot ok to test

**Comment by cmluciano** [comment]
@jiayingz The bazel build is claiming a merge conflict ```CONFLICT (content): Merge conflict in pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go```

File: pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go [hunk 1]
```diff
@@ -17,13 +17,15 @@ limitations under the License.
...
 import (
+	"reflect"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 
 	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/kubelet/dockershim/libdocker"
```
Comment (mindprince) [inline pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go]: Please run `go fmt` on this file. This seems unordered.

**Review by rohitagarwal003 — CHANGES_REQUESTED** [review]
`pkg/kubelet/gpu/nvidia/BUILD` needs to be updated with a new test dependency on:

```
//pkg/kubelet/dockershim/libdocker:go_default_library
```

File: pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go [hunk 2]
```diff
@@ -60,6 +62,34 @@ func makeTestPod(numContainers, gpusPerContainer int) *v1.Pod {
...
+
+	// Expects zero capacity without any GPUs.
+	gpuCapacity := testGpuManager2.Capacity()
+	as.Equal(len(gpuCapacity), 1)
+	rgpu := gpuCapacity[v1.ResourceNvidiaGPU]
+	as.Equal(rgpu.Value(), int64(0))
+
+	err2 := testGpuManager2.Start()
+	if err2 != nil {
+		gpus := reflect.ValueOf(testGpuManager2).Elem().FieldByName("allGPUs").Len()
```
Comment (cmluciano) [inline pkg/kubelet/gpu/nvidia/nvidia_gpu_manager_test.go]: Can you put a note of how many we do expect as a comment? Also will testGpuManager2.Start try to go through the discovery logic or do we want to hardcode it more?

**Pull request merged**